File: implementations/vb/Loom.Core/Models/ToolContext.vb
Namespace Models
    ''' <summary>
    ''' Which tools are available for the invoked type
    ''' </summary>
    Public Class ToolContext
        Public Property RegisteredTools As New List(Of ToolDefinition)
        Public Property MaxToolDepth As Integer = 5 ' prevents eternal loop calls
    End Class
End Namespace

File: implementations/csharp/src/Loom.Core/Models/ToolContext.cs
using System.Collections.Generic;

namespace Loom.Core.Models
{
    /// <summary>
    /// Which tools are available for the invoked type
    /// </summary>
    public class ToolContext
    {
        public List<ToolDefinition> RegisteredTools { get; set; } = new List<ToolDefinition>();
        public int MaxToolDepth { get; set; } = 5; // prevents eternal loop calls
    }
}
